src/main/kotlin/com/jetbrains/snakecharm/codeInsight/wrapper/WrapperStorage.kt

```diff
@@ -0,0 +1,59 @@
+package com.jetbrains.snakecharm.codeInsight.wrapper
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.util.xmlb.XmlSerializerUtil
+
+@State(name = "WrapperStorage", storages = [Storage("wrapper.xml")])
+class WrapperStorage : PersistentStateComponent<WrapperStorage> {
+
+    var wrapperStringStorage: MutableList<String> = mutableListOf()
+
+    data class Wrapper(
+            val repositoryTag: String,
+            val pathToWrapperDirectory: String,
+            val environmentFileContent: String,
+            val metaFileContent: String,
+            val testSnakefileContent: String
+    ) {
+        fun toList(): List<String> = listOf(
+                repositoryTag,
+                pathToWrapperDirectory,
+                environmentFileContent,
+                metaFileContent,
+                testSnakefileContent
+        )
+    }
+
+    companion object {
+        fun getInstance(): WrapperStorage = ApplicationManager.getApplication().getComponent(WrapperStorage::class.java)
+    }
+
+    fun addWrapper(wrapper: Wrapper) = wrapperStringStorage.addAll(wrapper.toList())
+
+    fun getWrapperList(): List<Wrapper> {
+        val wrappers = mutableListOf<Wrapper>()
+        // TODO can this be modified? storing everything as a list of strings is not preferable
+        val wrapperSize = 5
+        for (i in 0 until wrapperStringStorage.size step wrapperSize) {
+            wrappers.add(Wrapper(
+                    wrapperStringStorage[i],
+                    wrapperStringStorage[i + 1],
+                    wrapperStringStorage[i + 2],
+                    wrapperStringStorage[i + 3],
+                    wrapperStringStorage[i + 4]
```

Comment:
Rather dangerous place, you assume that `wrapperStringStorage` has enough lines.  and even doesn't handle AIOBE here. Not sure what is going on, but my error from SnakeCharm (after 1st or 2nd launch) is related to this

```
java.lang.IllegalArgumentException: Parameter specified as non-null is null: method com.jetbrains.snakecharm.codeInsight.wrapper.WrapperStorage$Wrapper.<init>, parameter repositoryTag
	at com.jetbrains.snakecharm.codeInsight.wrapper.WrapperStorage$Wrapper.<init>(WrapperStorage.kt)
	at com.jetbrains.snakecharm.codeInsight.wrapper.WrapperStorage.getWrapperList(WrapperStorage.kt:41)
	at com.jetbrains.snakecharm.codeInsight.completion.SmkWrapperCompletionProvider.addCompletions(SmkWrapperCompletionContributor.kt:41)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	...
```